### ISSUE TITLE
`no-named-as-default` and `no-named-as-default-member` should not be upset by TS `namespace` members

### ISSUE THREAD

**Issue by andersk** [un-ts/eslint-plugin-import-x#491]
Zulip sees several false positives from `import-x/no-named-as-default`:

```ts
import webpack from "webpack";
// warning  Using exported name 'webpack' as identifier for default export  import-x/no-named-as-default
```

and `import-x/no-named-as-default-member`:

```ts
import autosize from "autosize";
autosize.update(…);
// warning  Caution: `autosize` also has a named export `update`. Check if you meant to write `import {update} from 'autosize'` instead  import-x/no-named-as-default-member
autosize.destroy(…);
// warning  Caution: `autosize` also has a named export `destroy`. Check if you meant to write `import {destroy} from 'autosize'` instead  import-x/no-named-as-default-member

import ErrorStackParser from "error-stack-parser";
ErrorStackParser.parse(…);
// warning  Caution: `ErrorStackParser` also has a named export `parse`. Check if you meant to write `import {parse} from 'error-stack-parser'` instead  import-x/no-named-as-default-member

import assert from "minimalistic-assert";
assert.equal(…);
// warning  Caution: `assert` also has a named export `equal`. Check if you meant to write `import {equal} from 'minimalistic-assert'` instead  import-x/no-named-as-default-member
```

In reality, none of the named exports listed in these error messages exist; a default `import` is the _only_ way to consume these modules from ESM.

```js
> Object.keys(await import('webpack'))
[ 'default', 'module.exports' ]
> Object.keys(await import('autosize'))
[ 'default', 'module.exports' ]
> Object.keys(await import('error-stack-parser'))
[ 'default', 'module.exports' ]
> Object.keys(await import('minimalistic-assert'))
[ 'default', 'module.exports' ]
```

These modules are declared in TypeScript using an `export = …` assignment with a `namespace` (sometimes a `namespace` that’s been declaration-merged into a `function`, but that detail is immaterial here). These rules are considering members of the `namespace` to be named exports of the module, but shouldn’t, because they cannot be consumed via named `import` (unless the consumer is being transpiled to CommonJS).
